bvm/ballerina-core/src/main/java/org/ballerinalang/model/util/XMLUtils.java

```diff
@@ -335,7 +335,11 @@ public static BXML tableToXML(BTable table) {
      */
     public static BXML<?> createXMLText(String content) {
         // Remove carriage return on windows environments to eliminate additional &#xd; being added
-        content = content.replace("\r\n", "\n");
+        content = content
+                .replace("\r\n", "\n")
+                .replace("&gt;", ">")
```

Comment:
<, >, & symbols even though represented as escaped characters in xml (as &lt; &gt; &amp), to xml infoset they are just the plain character (<, >, &). And they are counted  as one character each.